app/res/xml/preferences_developer.xml

```diff
@@ -85,6 +85,11 @@
         android:entryValues="@array/pref_enabled_vals"
         android:key="cc-auto-purge"
         android:title="Detect and Trigger Purge on Form Save"/>
+    <EditTextPreference
```

Comment:
Instead of being an `EditTextPreference`, this should be a preference that when clicked, opens a file browser for the user to select a local file which contains a payload. Then when the user selects a file, CommCare saves the file path to the preference to be retrieved later. It can also do some validation to check whether you've selected a valid file type (a .xml file), and can reject the selection if it isn't. [The preference for `print-doc-location`](https://github.com/dimagi/commcare-android/blob/c3602a949e2fc753e53fa46a9390576fa3259154/app/src/org/commcare/preferences/CommCarePreferences.java#L83-L83) follows this exact model, so you can use that as an example.